addons/web/static/src/js/views/search_menus.js

```diff
@@ -231,6 +232,23 @@ return Widget.extend({
         }
         this.query.reset([this.facet_for(filter)], {
             preventSearch: preventSearch || false});
+
+        // Load sort settings on view
+        var self = this;
+        var sort_items = JSON.parse(filter.sort);
+        if (sort_items) {
+            _.each(sort_items.reverse(), function(sort_order) {
+                var sort_desc = false
+                if (sort_order[0] === '-') {
+                    sort_order = sort_order.slice(1)
+                    sort_desc = true
+                }
+                self.searchview.dataset.sort(sort_order)
```

Comment:
This will not work, the `sort` method toggles the order on a column. So if you were already sorting by `name ASC`, using a filter with `['name']` as filter will give you `name DESC` as filter.